controlpanel/oidc.py

```diff
@@ -0,0 +1,44 @@
+from urllib.parse import urlencode
+
+from django.conf import settings
+from django.urls import reverse
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+
+from controlpanel.api.models import User
+
+
+class OIDCSubAuthenticationBackend(OIDCAuthenticationBackend):
+    """
+    Authentication backend which matches users by their `sub` claim
+    """
+
+    def create_user(self, claims):
+        user = User.objects.create(
+            pk=claims.get("sub"),
+            username=claims.get(settings.OIDC_FIELD_USERNAME),
```

Comment:
OIDC_FIELD_USERNAME isn't in claims when I try to log in causing this to fail. Not sure if it's something weird with my user or if this code path hasn't been tested

Reply:
I've tested this, and it works for me - it's possible your Github user doesn't have a `nickname` claim (the value of `OIDC_FIELD_USERNAME`), but it seems unlikely